### ISSUE TITLE
Update JUnit version

### ISSUE THREAD

**Issue by furlaneto** [unitsofmeasurement/unit-api#71]
Hey guys, I saw that the **API** are not using the last version of **JUnit** and as improvement process, I would update the JUnit library.
I can start with the API and after the others projects like **indriya**

**Comment by keilw** [comment]
Sure, you worked on JNoSQL it seems, so I assume you signed the ECA or are Eclipse committer already? What about JCP membership? Both the JSR 385 API and RI Indriya require those who contribute to either be full or Associate JCP members. You don't seem to be right now (https://jcp.org/en/participation/members/F), but @otaviojava can certainly help you. And if some of JNoSQL migrates into a JCP standard some day, that could also help with that ;-)

**Comment by furlaneto** [comment]
Yeah, I signed the ECA.
Ok! I gonna see about JCP membership.
Thanks @keilw

**Comment by Daniel-Dos** [comment]
Hi,

here is update to version Junit 5 ?

**Comment by Daniel-Dos** [comment]
Hi,

yes, I see what the api still this with version junit 4.12 .

In this case, do we update to version 5?

**Comment by keilw** [comment]
It's quite an effort because all Unit tests need to be upgraded, some (especially those with expected exceptions) rewritten, but please  propose a PR (ideally after https://github.com/unitsofmeasurement/unit-api/pull/111 can be dealt with) for all the necessary changes.

**Comment by keilw** [comment]
@Daniel-Dos, Although @furlaneto may not be a JCP Associate Member yet, having signed the ECA towards Eclipse is a good first step, maybe you could collaborate on that?

**Comment by Daniel-Dos** [comment]
Hi ,

wonder,  I await the resolution the issue #111 

@furlaneto  you have me help whit this  ?

thanks.

**Comment by furlaneto** [comment]
@keilw now I'm a JCP member.

@Daniel-Dos I can help you

**Comment by Daniel-Dos** [comment]
wonder @furlaneto , when do you think we can play this issue?

thanks.

**Comment by thodorisbais** [comment]
I'd like to pick this up before we celebrate the first anniversary of this issue😝 
I'm completely new to JUnit 5, however, I already see that the same kind of work has been done for indriya as well; thus, I have a starting point, which is good.

On the other hand, I'm a bit concerned about @keilw 's reference above:
> all Unit tests need to be upgraded, some (especially those with expected exceptions) rewritten

So, the question here is: why do we need to update our tests (Java code) as well?
I see that by just using the `junit-vintage-engine` dependency (on top of the rest of required ones for JUnit 5), we don't have to change anything else (Java related).

That is, by only adding the dependencies we added for JUnit 5 migration for indriya, I got some test errors. I then just added `junit-vintage-engine` dependency as well and the tests passed again.

Isn't that what we want or am I just missing something?

Why do we need to rewrite/adjust tests for JUnit 5?
